### PR TITLE
added inline docs

### PR DESCRIPTION
**Pull request by nradz**
I added inline docs to the project. After this merge, I will created the first package to upload to Pypi.

### CODE REVIEW

File: RecordMapper/RecordMapper.py [hunk 1]
```diff
@@ -29,25 +41,54 @@ def __init__(self, base_schema: dict, nested_schemas: List[dict] = []):
...
-        self.transform_applier = TransformApplier()
-        self.clean_applier = CleanApplier()
+        self.selector_applier = NestedSchemaSelectorApplier(self.flat_schemas, self.custom_variables)
+        self.rename_applier = RenameApplier(self.custom_variables)
+        self.transform_applier = TransformApplier(self.custom_variables)
+        self.clean_applier = CleanApplier(self.custom_variables)
     
     def execute(self, input_format: str, input_file_path: str, paths_to_write: dict,
         input_opts: dict = {}, base_schema_to_write: dict = None, nested_schemas_to_write : List[dict] = None):
+        """The execute process that reads, transforms a writed the data from a format to another one.
```
Comment (ju4nlu) [inline RecordMapper/RecordMapper.py:51]: The execute process that reads, transforms **and writes** the data from a format to another one.

File: RecordMapper/RecordMapper.py [hunk 2]
```diff
@@ -29,25 +41,54 @@ def __init__(self, base_schema: dict, nested_schemas: List[dict] = []):
...
+        :type record_list: Iterable[dict]
+        :yield: A transformed record. 
+        :rtype: Iterator[dict]
+        """
 
         for record in record_list:
             yield self.transform_record(record)
 
     def transform_record(self, record: dict) -> dict:
+        """Apply transforms on a single record. Each transform step is defined in a Applier object.
```
Comment (ju4nlu) [inline RecordMapper/RecordMapper.py:85]: ...  in **an** Applier object

File: RecordMapper/RecordMapper.py [hunk 3]
```diff
@@ -68,7 +109,21 @@ def transform_record(self, record: dict) -> dict:
 
         return normal_record
     
-    def read_records(self, input_format: str, path_to_read: str, opts : dict = {}):
+    def read_records(self, input_format: str, path_to_read: str, opts: dict = {}) -> Iterator[dict]:
+        """Read records of a input file.
```
Comment (ju4nlu) [inline RecordMapper/RecordMapper.py:113]: ... of **an** input file

File: RecordMapper/builders/FunctionBuilder.py [hunk 4]
```diff
@@ -31,7 +39,17 @@ def parse_function_str(function_str: str) -> Callable[[object, dict], object]:
  
 
     @staticmethod
-    def get_builtin_function(function_name: str, args_list: List[str]):
+    def get_builtin_function(function_name: str, args_list: List[str]) -> Callable[[object, dict], object]:
+        """Get a builtin function from a string.
```
Comment (eoterogarcia) [inline RecordMapper/builders/FunctionBuilder.py:43]: _build-in_

File: RecordMapper/builders/FunctionBuilder.py [hunk 5]
```diff
@@ -42,6 +60,18 @@ def get_builtin_function(function_name: str, args_list: List[str]):
 
     @staticmethod
     def get_custom_function(function_name: str, args_list: List[str]):
+        """Get a custom function from a string. This string is a complete import path to the function.
```
Comment (eoterogarcia) [inline RecordMapper/builders/FunctionBuilder.py:63]: This string is a complete import path _of_ the function.

File: RecordMapper/csv/CSVReader.py [hunk 6]
```diff
@@ -1,17 +1,30 @@
 import csv
-from typing import BinaryIO
-
+from typing import BinaryIO, Iterator
 
 from RecordMapper.common import Reader
 
 class CSVReader(Reader):
+    """A object that read records form a input csv file.
```
Comment (eoterogarcia) [inline RecordMapper/csv/CSVReader.py:7]: _Reads records from an input csv file_

File: RecordMapper/csv/CSVWriter.py [hunk 7]
```diff
@@ -1,25 +1,50 @@
 import csv
 import json
-from typing import BinaryIO, List
+from typing import BinaryIO, List, Iterable
 
 from RecordMapper.common import Writer
 
 class CSVWriter(Writer):
+    """The object that write the records to a csv file.
```
Comment (eoterogarcia) [inline RecordMapper/csv/CSVWriter.py:8]: _Writes the records to a csv file_

File: RecordMapper/appliers/TransformApplier.py [hunk 8]
```diff
@@ -5,10 +5,31 @@
 from typing import Callable, List
 
 
-
 class TransformApplier(object):
+    """This applier execute the defined transformations for each field.
```
Comment (ju4nlu) [inline RecordMapper/appliers/TransformApplier.py:9]: **executes**

File: RecordMapper/csv/CSVWriter.py [hunk 9]
```diff
@@ -1,25 +1,50 @@
...
+            Their order will be preserved in the result file.
+        :type fieldnames: List[str]
+        """
+
+        super().__init__(file_path)
         self.fieldnames = fieldnames
 
-    def write_records_to_output(self, records: List[dict], output: BinaryIO):
+    def write_records_to_output(self, records: Iterable[dict], output: BinaryIO):
+        """Write the records to an output file.
```
Comment (eoterogarcia) [inline RecordMapper/csv/CSVWriter.py:25]: _Writes_

File: RecordMapper/csv/CSVWriter.py [hunk 10]
```diff
@@ -1,25 +1,50 @@
...
 
         csv_writer = csv.DictWriter(output, fieldnames=self.fieldnames)
         csv_writer.writeheader()
 
         for record in records:
             csv_writer.writerow(self.format_record(record))
     
-    def format_record(self, record: dict):
+    def format_record(self, record: dict) -> dict:
+        """Format a record to be writed. For example, dicts will be serialized to
```
Comment (eoterogarcia) [inline RecordMapper/csv/CSVWriter.py:40]: _to be written_

File: RecordMapper/avro/AvroReader.py [hunk 11]
```diff
@@ -1,18 +1,32 @@
...
+        :type path_to_read: string
+        """
 
         super().__init__(path_to_read)
 
         self.read_options ="rb"
 
-    def read_records_from_input(self, input_stream: BinaryIO):
+    def read_records_from_input(self, input_stream: BinaryIO) -> Iterator[dict]:
+        """Read records from a input stream in avro format.
```
Comment (ju4nlu) [inline RecordMapper/avro/AvroReader.py:23]: **an**

File: RecordMapper/builders/BuiltinFunctions.py [hunk 12]
```diff
@@ -1,23 +1,33 @@
+"""
+This module defines a set of built-in functions.
+"""
 
 def copyFrom(path_to_copy_from: str):
+    """This built-in function return the value of 'path_to_copy_from' key.
```
Comment (ju4nlu) [inline RecordMapper/builders/BuiltinFunctions.py:6]: **returns**

File: RecordMapper/csv/CSVWriter.py [hunk 13]
```diff
@@ -1,25 +1,50 @@
...
+        JSON strings.
+
+        :param record: A input record.
+        :type record: dict
+        :return: A formatted record to be writed.
```
Comment (eoterogarcia) [inline RecordMapper/csv/CSVWriter.py:45]: _to be written_

File: RecordMapper/builders/BuiltinFunctions.py [hunk 14]
```diff
@@ -1,23 +1,33 @@
...
+    """
 
-    def transform_function(current_value: object, record: dict, complete_transform_schema: dict, is_nested_record: bool=False):
+    def transform_function(current_value: object, record: dict, complete_transform_schema: dict, custom_variables: dict, is_nested_record: bool=False):
         return None
 
     return transform_function
 
 def toInt():
+    """This built-in function cast the current value to Int and returns the result.
```
Comment (ju4nlu) [inline RecordMapper/builders/BuiltinFunctions.py:26]: **casts**

File: RecordMapper/builders/FlatRecordBuilder.py [hunk 15]
```diff
@@ -1,13 +1,47 @@
...
+        :param record: The input record.
+        :type record: dict
+        :return: A flat record.
+        :rtype: dict
+        """
         return dict([((key,), value) for key, value in record.items()])
     
     @staticmethod
     def get_normal_record_from_flat_record(flat_record: dict) -> dict:
+        """
```
Comment (ju4nlu) [inline RecordMapper/builders/FlatRecordBuilder.py:38]: This inline doc looks shifted to the righ

File: RecordMapper/builders/FlatRecordBuilder.py [hunk 16]
```diff
@@ -1,13 +1,47 @@
...
+    {
+        ("field_1",): 5,
+        ("field_2","field_x"): 7,
+        ("field_2","field_y"): 9
+    }
+    """
 
     @staticmethod
     def get_flat_record_from_normal_record(record: dict) -> dict:
+        """Flat a record.
```
Comment (ju4nlu) [inline RecordMapper/builders/FlatRecordBuilder.py:27]: **Flattens a record**

File: RecordMapper/builders/FunctionBuilder.py [hunk 17]
```diff
@@ -6,12 +6,20 @@
 from RecordMapper.builders import BuiltinFunctions
 
 class InvalidFunctionError(Exception):
+    """An exception that represents a invalid string reference for a function"""
```
Comment (ju4nlu) [inline RecordMapper/builders/FunctionBuilder.py:9]: **an** invalid string reference

File: RecordMapper/builders/FunctionBuilder.py [hunk 18]
```diff
@@ -6,12 +6,20 @@
...
     pass
 
 class FunctionBuilder(object):
+    """A Builder class of functions"""
 
     @staticmethod
     def parse_function_str(function_str: str) -> Callable[[object, dict], object]:
+        """This function parses a string and generates a function to be used as transform function
```
Comment (ju4nlu) [inline RecordMapper/builders/FunctionBuilder.py:17]: Add a new line to keep consistency with other functions inline docs

Comment (ju4nlu) [reply RecordMapper/builders/FunctionBuilder.py:17]: to be uses as **a** transform function

File: RecordMapper/utils.py [hunk 19]
```diff
@@ -14,7 +19,14 @@ def wrapper_function(*input_args):
     return wrapper_function
 
 
-def object_as_tuple(obj):
+def object_as_tuple(obj: object) -> tuple:
+    """Convert an object in a tuple (except if this object is a tuple).
```
Comment (eoterogarcia) [inline RecordMapper/utils.py:23]: except if this object is already a tuple

File: RecordMapper/csv/CSVReader.py (hunk 6 above)
Comment (ju4nlu) [inline RecordMapper/csv/CSVReader.py:7]: **An** object that **reads** records **from an** input csv file

File: RecordMapper/csv/CSVReader.py [hunk 20]
```diff
@@ -1,17 +1,30 @@
...
+        """
 
-    def __init__(self, input: object):
+        super().__init__(file_path)
+        self.filepath = file_path
 
-        super().__init__(input)
-        self.filepath = input
+    def read_records_from_input(self, input_stream: BinaryIO) -> Iterator[dict]:
+        """The function that read the records from the input stream.
```
Comment (ju4nlu) [inline RecordMapper/csv/CSVReader.py:21]: **This function reads records** from ...

File: RecordMapper/csv/CSVWriter.py (hunk 7 above)
Comment (ju4nlu) [inline RecordMapper/csv/CSVWriter.py:8]: **Object** that **writes records** to a csv file

File: RecordMapper/csv/CSVWriter.py (hunk 10 above)
Comment (ju4nlu) [inline RecordMapper/csv/CSVWriter.py:40]: Format a record to be **written**

File: RecordMapper/utils.py (hunk 19 above)
Comment (ju4nlu) [inline RecordMapper/utils.py:23]: **Transforms** an object **into** a tuple

File: RecordMapper/RecordMapper.py (hunk 1 above)
Comment (nradz) [reply RecordMapper/RecordMapper.py:51]: ok

File: RecordMapper/builders/FunctionBuilder.py (hunk 4 above)
Comment (nradz) [reply RecordMapper/builders/FunctionBuilder.py:43]: I think that built-in is correct -> https://www.yourdictionary.com/built-in-function#:~:text=built%2Din%2Dfunction-,built%2Din%2Dfunction,reference%20with%20specification%20of%20arguments

Comment (eoterogarcia) [reply RecordMapper/builders/FunctionBuilder.py]: True! you're right

**Pull request merged**